### PR TITLE
PMM-7218 Generate rule files

### PR DESCRIPTION
**Pull request by artemgavrilov**
[PMM-7218](https://jira.percona.com/browse/PMM-7218)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1353

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Comment by artemgavrilov** [comment]
Now Alerts API is working

File: services/management/ia/rules_service_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+				Value: "60",
+			},
+		},
+		ChannelIds: []string{channelID},
+	})
+	require.NoError(t, err)
+	ruleID := resp.RuleId
+	defer func() {
+		e := db.InTransaction(func(t *reform.TX) error {
+			return models.RemoveRule(t.Querier, ruleID)
```
Comment (AlekSi) [inline services/management/ia/rules_service_test.go:123]: `testdb.Open` recreates the whole database. There is no need to do manual cleanup.

**Review by AlekSi — COMMENTED** [review]
Please check two things:
* that we don't generated files for disabled rules – I don't see a check for that;
* that invalid built-in rules (for example, without pmm_ prefix) break tests – test that loads built-in files seems to be removed.

File: services/management/ia/rules_service_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            max_over_time(mysql_global_status_threads_connected[5m]) / ignoring (job)
+            mysql_global_variables_max_connections
+            * 100
+            > 1.22
+          for: 2s
+          labels:
+            baz: faz
+            foo: bar
+            ia: "1"
+            severity: SEVERITY_INFO
```
Comment (AlekSi) [inline services/management/ia/rules_service_test.go:149]: Both rule ID and template name should be there

File: services/management/ia/rules_service_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          annotations:
+            description: |-
+                More than 1.22%% of MySQL connections are in use on {{ $labels.instance }}
+                VALUE = {{ $value }}
+                LABELS: {{ $labels }}
+            rule_summary: some testing rule
```
Comment (AlekSi) [inline services/management/ia/rules_service_test.go:155]: Just "rule"

File: services/management/ia/rules_service.go [hunk 4]
```diff
@@ -17,38 +17,225 @@
...
+	rulesPath    string // used for testing
+
 }
 
 // NewRulesService creates an API for Integrated Alerting Rules.
-func NewRulesService(db *reform.DB, templates *TemplatesService, vmalert vmAlertService) *RulesService {
+func NewRulesService(db *reform.DB, templates *TemplatesService, vmalert vmAlert, alertManager alertManager) *RulesService {
+	l := logrus.WithField("component", "management/ia/rules")
+
+	err := dir.CreateDataDir(rulesDir, "pmm", "pmm", dirPerm)
```
Comment (palash25) [inline services/management/ia/rules_service.go:60]: you might want to add another call to change the owners of its parent dir `"/etc/ia/"` last time I was doing it, the owner:group of the parent dir turned out to be different from the child so I added two calls with separate paths in the templates service

File: services/management/ia/templates_service.go [hunk 5]
```diff
@@ -80,21 +70,7 @@ type TemplatesService struct {
 func NewTemplatesService(db *reform.DB) *TemplatesService {
 	l := logrus.WithField("component", "management/ia/templates")
 
-	err := dir.CreateDataDir(templatesParentDir, "pmm", "pmm", dirPerm)
```
Comment (palash25) [inline services/management/ia/templates_service.go:83]: same as https://github.com/percona/pmm-managed/pull/602/files#r548864528

File: services/management/ia/rules_service.go (hunk 4 above)
Comment (artemgavrilov) [reply services/management/ia/rules_service.go:60]: Does it caused some problems in you case? And did you find reason why parend dir had differerent owner?

File: services/management/ia/rules_service_test.go (hunk 1 above)
Comment (artemgavrilov) [reply services/management/ia/rules_service_test.go]: 🤦‍♂️ removed

File: services/management/ia/rules_service_test.go (hunk 2 above)
Comment (artemgavrilov) [reply services/management/ia/rules_service_test.go:139]: Added

File: services/management/ia/rules_service_test.go (hunk 3 above)
Comment (artemgavrilov) [reply services/management/ia/rules_service_test.go]: Renamed

File: services/management/ia/rules_service.go (hunk 4 above)
Comment (palash25) [reply services/management/ia/rules_service.go:60]: > Does it caused some problems in you case?

not that I can remember, for context you can take a look at this thread https://github.com/percona/pmm-managed/pull/558/#discussion_r536101878

> And did you find reason why parend dir had differerent owner?

no I couldn't

Comment (artemgavrilov) [reply services/management/ia/rules_service.go:60]: Well, lets keep it as is until there is no problems. But I will keep in mind that such case is possible.

**Review by palash25 — APPROVED** [review]

**Review by AlekSi — COMMENTED** [review]
API tests fail

**Comment by artemgavrilov** [comment]
> API tests fail

I recreated FB, and now API tests are green. Previous build was bad for some reason. 
Now there is another porblem.

File: services/management/ia/rules_service.go (hunk 4 above)
Comment (AlekSi) [reply services/management/ia/rules_service.go:60]: There should not be a problem for `/etc/…` dirs that are internal for container. `/srv/…`, however, can be on an external volume, and there problems could happen

**Review by AlekSi — APPROVED** [review]

**Pull request merged**